### ISSUE TITLE
Loss functions applied in alphabetical order instead of by dictionary keys (since Keras 3.5.0)

### ISSUE THREAD

**Issue by malcolmlett** [googlecolab/colabtools#4988]
**Current behavior**
When training models with multiple outputs, and using dictionary as `loss` parameter to model.compile(), the wrong loss functions are applied against the different model outputs. The alphabetical order of the dictionary keys determines the order of loss function against model output.

**Expected behavior**
Loss functions should be applied against named model outputs.

**Workaround**
Bug is with Keras 2.5.0 (Google Colab's current default), and can be avoided by downgrading to Keras 2.4.1.

**Additional context**
Issue has been raised with Keras team: https://github.com/keras-team/keras/issues/20596

This present issue is for awareness of Colab team.

**Comment by cperry-goog** [comment]
This is a great bug report, thank you! I'll pass on to the Keras folks as well.

**Comment by malcolmlett** [comment]
Update: I'd mistakenly typed Keras 2.5.0  and 2.4.1, when that should have been 3.5.0 / 3.4.1 -- have updated the description.

Also, Keras folks say that Keras 3.7.0 works ok. What does that mean for Google Colab and its current 3.5.0 default?

**Comment by colaboratory-team** [comment]
Tracking internally as b/383879003, we will try to get Keras updated

**Comment by metrizable** [comment]
Colab now comes pre-installed with `keras==3.8.0`
